internal/leadership/elector.go

```diff
@@ -254,6 +228,45 @@ func (e *Elector) attemptGainLeadershipLoop(ctx context.Context) error {
 	}
 }
 
+// Handles a leadership notification from the notifier.
+func (e *Elector) handleLeadershipNotification(ctx context.Context, topic notifier.NotificationTopic, payload string) {
+	if topic != notifier.NotificationTopicLeadership {
+		// This should not happen unless the notifier is broken.
+		e.Logger.Error(e.Name+": Received unexpected notification", "client_id", e.clientID, "topic", topic, "payload", payload)
+		return
+	}
+
+	notification := dbLeadershipNotification{}
+	if err := json.Unmarshal([]byte(payload), &notification); err != nil {
+		e.Logger.Error(e.Name+": Unable to unmarshal leadership notification", "client_id", e.clientID, "err", err)
+		return
+	}
+
+	e.Logger.InfoContext(ctx, e.Name+": Received notification from notifier", "action", notification.Action, "client_id", e.clientID)
+
+	if notification.Action != "resigned" || notification.Name != e.instanceName {
+		// We only care about resignations because we use them to preempt the
+		// election attempt backoff. And we only care about our own key name.
+		return
+	}
+
+	// If this a resignation from _this_ client, ignore the change.
+	if notification.LeaderID == e.clientID {
```

Comment:
There's a little noise in this PR because I extracted this function so I could test it more easily, but this is the relevant clause that was added here. I also added the early context check just below, but everything else just moves around and was preexisting.